### ISSUE TITLE
WordPress.Security.NonceVerification.Missing not reported when using filter_input

### ISSUE THREAD

**Issue by kkmuffme** [WordPress-Coding-Standards/WordPress-Coding-Standards#2299]
## Bug Description
When using filter_input with INPUT_GET or INPUT_POST or INPUT_COOKIE, it's like accessing the $_GET/$_POST superglobals.
Therefore a nonce validation is required.

## Minimal Code Snippet
The issue happens when running this command:
```bash
./bin/phpcs --config-set installed_paths ../wpcs
```

... over a file containing this code:
```php
// should report an error
$foo = filter_input( INPUT_POST, 'email', FILTER_SANITIZE_EMAIL );
update_option( 'abc', $foo, false );

// correctly reports an error:
$foo = $_POST['email'];
update_option( 'def', $foo, false );
```

## Error Code
WordPress.Security.NonceVerification.Missing

## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.4
| PHP_CodeSniffer version | 3.7.2
| WPCS version            | 2.3.0
| WPCS install type       | https://github.com/WordPress/WordPress-Coding-Standards#standalone

## Tested Against `develop` branch?
- [ ] I have verified the issue still exists in the `develop` branch of WPCS.

**Comment by johnstonphilip** [comment]
+1

**Comment by jrfnl** [comment]
@johnstonphilip Please don't spam issues with "+1"'s which don't contribute anything useful to the discussion. You can use the emoji response on the first post for that.

**Comment by johnstonphilip** [comment]
I didn't intend to spam. My apologies. I'm just wanting to note that this is a very important issue to me. Thanks, and my apologies again!
